### DISCUSSION TITLE
RxTutor: An Interactive sandbox for RxJS Observables

### DISCUSSION THREAD

**Discussion by lgpage** [ReactiveX/rxjs#7133] [Show and tell]
https://rxtutor.org/

An interactive sandbox for RxJS Observables

![image](https://user-images.githubusercontent.com/1577821/209124053-58d42f5b-1bea-4fdf-ba98-bc7c776651d2.png)

https://github.com/lgpage/rxtutor

**Comment by voliva** [comment]
Wow this is awsome! It's interactive like https://rxmarbles.com/ but you can also change the parameters and code the observable as you like.

Is it possible to use other operators than the one currently selected though? I see that if I'm on the `switchMap` page, I can only use `switchMap`

**Reply by lgpage** [reply]
You can use any operators from rxjs. Each example is meant as a starting point for change and playing around.
You just need to use the destructuring assignment syntax to add the operators you want. For example:
```js
function visualize({ switchMap, map, tap }, one$, two$) {
  return one$.pipe(
    switchMap((a) => two$.pipe(
      map((b) => `${a}${b}`),
      tap((x) => console.log({ x })),
    )),
  );
}
```

Or you can change the function like such:
```js
function visualize(rx, one$, two$) {
  return one$.pipe(
    rx.switchMap((a) => two$.pipe(
      rx.map((b) => `${a}${b}`),
      rx.tap((x) => console.log({ x })),
    )),
  );
}
```

**Reply by voliva** [reply]
Ohh I didn't realise the operators came from within the function, I tried using an import but that didn't work 😅 

Thank you!
